### DISCUSSION TITLE
running on macos

### DISCUSSION THREAD

**Discussion by mightgoyardstill** [tehn/isms#5] [General]
what type of considerations would you recommend if i was to try and run this on mac?

i understand there are some linux specific dependancies for audio routing?

is supercollider built with the application or is it run as a background process?

J

**Comment by p3r7** [comment]
you might get a better answer from Brian, but i guess i could share what i have discovered.

> i understand there are some linux specific dependancies for audio routing?

i guess so, it depends on ALSA which i believe is Linux-only. 
but this library is only used for MIDI I/O here, so i guess it wouldn't be tremendously hard to support other OSes w/ platform-specific implem for this specific part of the code.

idk if some of the other required libs are Linux-only, though...

> is supercollider built with the application or is it run as a background process?

i believe you have to launch the supercollider server yourself, isms doesn't spawn it for you.
your isms scrips can communicate w/ supercollider by [sending osc messages](https://github.com/tehn/isms/blob/main/example.lua#L27) (which is also how Supercollider IDE interacts w/ its server).

**Comment by tehn** [comment]
midi would take effort to be portable, for the reason mentioned. i'm not terribly familiar with working with midi on macos or windows.

device detection in general is very linux-specific as udev figures in prominently. for multi-platform grid detection, serialosc provides a model though to me it's far too complex.

this project itself does no audio. think of it as a lua environment with support for grids, midi, osc, and screen/keyboard (via sdl).

unless you're interested in actively developing cross-platform capabilities (which is an engineering challenge) i might recommend trying it out on a raspberry pi, which should work fine.
